build_tools/create_log_index.py

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python3
+
+import os
+import subprocess
+from pathlib import Path
+import sys
+
+
+def log(*args):
+    print(*args)
+    sys.stdout.flush()
+
+
+def index_log_files(build_dir: Path, amdgpu_family: str):
+    log_dir = build_dir / "logs"
+    index_file = log_dir / "index.html"
+
+    if log_dir.is_dir():
+        log(f"[INFO] Found '{log_dir}' directory. Indexing '*.log' files...")
+        subprocess.run(
+            ["python", str(build_dir / "indexer.py"), "-f", "*.log", str(log_dir)],
+            check=True,
+        )
+    else:
+        log(f"[WARN] Log directory '{log_dir}' not found. Skipping indexing.")
+        return
+
+    if index_file.exists():
+        log(
+            f"[INFO] Rewriting links in '{index_file}' with AMDGPU_FAMILIES={amdgpu_family}..."
+        )
+        content = index_file.read_text()
+        updated = content.replace(
+            'a href=".."', f'a href="../../index-{amdgpu_family}.html"'
+        )
```

Comment:
Relating to my other comment, why do we rewrite these index file names? Aren't they already in separate subdirectories for each configuration?